pkg/networking/networking.go

```diff
@@ -1,77 +1,63 @@
 package networking
 
 import (
-	"crypto/tls"
 	"crypto/x509"
 	"io"
 	"log"
 	"net/http"
 	"net/url"
 	"os"
+	"strings"
 
-	"github.com/snyk/go-application-framework/internal/constants"
+	"github.com/snyk/go-application-framework/pkg/auth"
 	"github.com/snyk/go-application-framework/pkg/configuration"
 	"github.com/snyk/go-application-framework/pkg/networking/certs"
+	"github.com/snyk/go-application-framework/pkg/networking/middleware"
 	"github.com/snyk/go-httpauth/pkg/httpauth"
 )
 
 //go:generate $GOPATH/bin/mockgen -source=networking.go -destination ../mocks/networking.go -package mocks -self_package github.com/snyk/go-application-framework/pkg/networking/
 
-const (
-	defaultUserAgent string = "snyk-cli"
-)
-
 // NetworkAccess is the interface for network access.
 type NetworkAccess interface {
-	// GetDefaultHeader returns the default header for a given URL.
-	GetDefaultHeader(url *url.URL) http.Header
-	// GetRoundtripper returns the http.Roundtripper.
-	GetRoundtripper() http.RoundTripper
+	// AddHeaders adds all the custom and authentication headers to the request.
+	AddHeaders(request *http.Request) error
+	// AddDefaultHeader adds the default headers request.
+	AddDefaultHeader(request *http.Request)
```

Comment:
Question: It appears we only use this method internally. How about removing it from the public API to keep the API slim?

Reply:
this would have the benefit we wouldn't need to explain to anyone the difference between AddHeader() and AddDefaultHeader()